Tests/HTTPTests/HTTPClientTests.swift

```diff
@@ -37,6 +37,26 @@ class HTTPClientTests: XCTestCase {
     func testQuery() throws {
         try testURL("http://httpbin.org/get?foo=bar", contains: "bar")
     }
+    
+    func testHeaderDecoration() throws {
```

Comment:
Instead of making `HTTPClientRequestSerializer` internal and testing it directly, we could query `http://httpbin.org/anything` using `HTTPClient` and check the returned request headers. 

This might be better since it minimizes changes and is more holistic.

Reply:
Done :)